### ISSUE TITLE
[prefer-readonly-parameter-types] Possible false positive?

### ISSUE THREAD

**Issue by OmgImAlexis** [typescript-eslint/typescript-eslint#1876]
**What were you trying to do?**

```json
{
  "rules": {
    "@typescript-eslint/prefer-readonly-parameter-types": ["error"]
  }
}
```

```ts
interface PluginModule {
	readonly filePath: string;
	readonly isActive: boolean;
}

type PluginModuleWithName = PluginModule & {
	readonly name: string;
};

interface Plugin {
	readonly isActive: boolean;
	readonly modules: Array<Readonly<PluginModule | PluginModuleWithName>>;
	readonly disabled?: string;
}

interface SetActiveStateOptions {
	readonly plugin: Readonly<Plugin>;
	readonly pluginName: string;
	readonly moduleName: string;
	readonly active: boolean;
}

const setActiveState = (
	options: Readonly<SetActiveStateOptions>
) => {
	console.log(options);
};
```

**What did you expect to happen?**
No error.

**What actually happened?**
```bash
✖  24:2  Parameter should be a read only type                @typescript-eslint/prefer-readonly-parameter-types
```

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`      | `2.26.0` |
| `@typescript-eslint/parser`             | `2.26.0 ` |
| `@typescript-eslint/typescript-estree`  | `2.26.0` |
| `@typescript-eslint/experimental-utils` | `2.26.0` |
| `TypeScript`                            | `3.8.3` |
| `node`                                  | `v12.5.0` |
| `npm`                                   | `6.13.0` |

**Comment by OmgImAlexis** [comment]
I'm not really sure what else I can do to make `options` "more" readonly. Am I missing something?
